### PR TITLE
Introduce ReferenceCell::n_isotropic_children().

### PR DESCRIPTION
**Pull request by bangerth**
I had a need to query the number of children of a cell, but it turns out that there is currently no function in `ReferenceCell` to do that. So I added one, but this is WIP because I don't know what we do (or intend to do) for refinement of tetrahedra and pyramids. Anyone remember the correct numbers to be added there?

/rebuild

### CODE REVIEW

**Comment by drwells** [comment]
We are using 8 children per tet: see https://github.com/dealii/dealii/blob/master/source/grid/tria.cc#L5127. This PR would be a nice place to un-hardcode some of those numbers.

I don't think we have figured out how to refine pyramids yet so making that fail is the right choice.

**Comment by bangerth** [comment]
Maybe use it like in that last commit?

**Comment by bangerth** [comment]
I pushed one more commit that cleans up a piece of code that wasn't immediately clear to me. I *think* that what it wants to test is that a number if zero/not-zero. I added an assertion that provides us with another nice use of the new function.

**Review by drwells — APPROVED** [review]

**Comment by bangerth** [comment]
The last commit failed; the new version of it should now succeed. It illustrates why we decided to deprecate `number_of_children()` and should be using `n_children()` instead :-)

File: include/deal.II/grid/reference_cell.h [hunk 1]
```diff
@@ -1019,25 +1040,64 @@ ReferenceCell::n_faces() const
...
+  if (*this == ReferenceCells::Vertex)
+    return 0;
+  else if (*this == ReferenceCells::Line)
+    return 2;
+  else if (*this == ReferenceCells::Triangle)
+    return 4;
+  else if (*this == ReferenceCells::Quadrilateral)
+    return 4;
+  else if (*this == ReferenceCells::Tetrahedron)
+    return 9;
```
Comment (drwells) [inline include/deal.II/grid/reference_cell.h:1062]: Isn't this 8 in line 5127 of tria.cc?

**Review by drwells — CHANGES_REQUESTED** [review]
I am pretty sure that should be an 8 - lets hold off on merging.

File: include/deal.II/grid/reference_cell.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/grid/reference_cell.h:1062]: Indeed.

**Comment by bangerth** [comment]
I fatfingered the number that @drwells provided for tetrahedra in a comment above. Sorry about that. It's fixed here now.

**Review by drwells — APPROVED** [review]

**Pull request merged**